app/controllers/person/security_tools_controller.rb

```diff
@@ -0,0 +1,118 @@
+#  Copyright (c) 2012-2022, Pfadibewegung Schweiz. This file is part of
+#  hitobito and licensed under the Affero General Public License version 3
+#  or later. See the COPYING file at the top-level directory or at
+#  https://github.com/hitobito/hitobito.
+
+class Person::SecurityToolsController < ApplicationController
+
+  PASSWORD_COMPROMISED_SITUATION = 'password_compromised_situation'
+  PASSWORD_COMPROMISED_SOLUTION = 'password_compromised_solution'
+  EMAIL_COMPROMISED_SITUATION = 'email_compromised_situation'
+  EMAIL_COMPROMISED_SOLUTION = 'email_compromised_solution'
+  DATALEAK_SITUATION = 'dataleak_situation_id'
+  DATALEAK_SOLUTION = 'dataleak_solution_id'
+  SUSPEND_PERSON_SITUATION = 'suspend_person_situation_id'
+  SUSPEND_PERSON_SOLUTION = 'suspend_person_solution_id'
+
+  before_action :authorize_action
+
+  decorates :group, :person, :security_tools
+
+  helper_method :person_has_login, :person_has_two_factor, :person, :herself, :group,
+  :password_compromised_situation_text, :password_compromised_solution_text,
+  :email_compromised_situation_text, :email_compromised_solution_text,
+  :dataleak_situation_text, :dataleak_solution_text,
+  :suspend_person_situation_text, :suspend_person_solution_text
+
+  respond_to :html
+
+  def password_override
+    person.encrypted_password = nil
+    person.save
+    notify
+    if herself
+      sign_out_and_redirect(current_user)
+    else
+      redirect_to group_person_path(group, person), notice: t('.flashes.success')
+    end
+  end
+
+  def person_has_two_factor
+    return person.two_factor_authentication
+  end
+
+  def person_has_login
+    return person.email?
+  end
```

Comment:
Solche Methoden würde ich eher direkt auf dem Model (oder einem entsprechenden Model Concern) definieren und in der View dann auf die `person` callen.
Dadurch können wir die Anzahl helper_method's etwas reduzieren.

Reply:
Habe ich entsprechend angepasst